### PR TITLE
Add SSL_CTX_set1_groups_list and SSL_set1_groups_list APIs

### PR DESCRIPTION
**Pull request by julek-wolfssl**
Based on a feature request in #2766

### CODE REVIEW

**Comment by dgarske** [comment]
Retest this please

File: src/tls13.c [hunk 1]
```diff
@@ -8218,6 +8218,85 @@ int wolfSSL_request_certificate(WOLFSSL* ssl)
 }
 #endif /* !NO_CERTS && WOLFSSL_POST_HANDSHAKE_AUTH */
 
+
+#if defined(OPENSSL_EXTRA) && defined(HAVE_ECC)
```
Comment (dgarske) [inline src/tls13.c]: Typically we put the compatibility functions in ssl.c. Perhaps move to below `wolfSSL_EC_curve_nid2nist`?

File: src/tls13.c [hunk 2]
```diff
@@ -8218,6 +8218,85 @@ int wolfSSL_request_certificate(WOLFSSL* ssl)
...
+                (len > kNistCurves_MAX_NAME_LEN)) {
+            WOLFSSL_MSG("Unrecognized curve name in list");
+            return -1;
+        }
+        for (nist_name = kNistCurves; nist_name->name != NULL; nist_name++) {
+            if (XSTRNCMP(list, nist_name->name, nist_name->name_len) == 0) {
+                break;
+            }
+        }
+        if (!nist_name->name_len) {
```
Comment (dgarske) [inline src/tls13.c]: The for loop above uses `nist_name->name != NULL` and this uses `!nist_name->name_len`. Was that on purpose?

File: src/ssl.c [hunk 3]
```diff
@@ -134,6 +134,34 @@
 #endif /* !WOLFCRYPT_ONLY || OPENSSL_EXTRA */
 #ifndef WOLFCRYPT_ONLY
 
+#if defined(OPENSSL_EXTRA) && defined(HAVE_ECC)
+const WOLF_EC_NIST_NAME kNistCurves[] = {
+    {STR_SIZEOF("P-192"),   "P-192",   NID_X9_62_prime192v1},
```
Comment (dgarske) [inline src/ssl.c]: I like the `STR_SIZEOF` macro use, but perhaps put it here in the .c file and only for this OPENSSL_EXTRA and HAVE_ECC case? Concerned this macro name is too generic and may collide with customer macros.

File: src/tls13.c (hunk 2 above)
Comment (julek-wolfssl) [reply src/tls13.c]: You are right. This was a mistake.

File: src/ssl.c (hunk 3 above)
Comment (julek-wolfssl) [reply src/ssl.c]: I put the macro definition in `wolfssl/internal.h` so it isn't included in the final build. I don't think there is a possibility of a collision.

**Pull request merged**